packages/web-client/app/utils/wc-provider.ts

```diff
@@ -517,8 +519,35 @@ class WalletConnectProvider extends ExtendedProviderEngine {
     };
   }
 
+  bindSocketListeners() {
+    this.websocketProvider.on('close', this.onWebsocketClose.bind(this));
+    this.websocketProvider.on('connect', this.onWebsocketConnect.bind(this));
+  }
+
+  async maybeReconnect() {
+    setTimeout(() => {
```

Comment:
How come `setTimeout` is needed?

Reply:
`setTimeout` is needed to delay execution of this code until the websocket provider has finished the callback for its close event, because part of that callback includes clearing all event listeners attached to it. We need to reattach the event listeners after that completes.

Reply:
This deserves to be a code comment.